examples/lang2/erc721/src/lib.rs

```diff
@@ -0,0 +1,269 @@
+        fn new(&mut self) {
+            self.none_account.set(AccountId::from([0x0; 32]));
+        }
+
+        /// ===========================
+        /// NOT REQUIRED, JUST FOR TEST
+        /// ===========================
+        #[ink(message)]
+        fn mint(&mut self, id: Hash) -> bool {
+            let caller = self.env().caller();
+            let owner = self.owner_of_or_none(id);
+            if owner != *self.none_account {
+                self.env().println(&format!("Erc721::mint::token is exist"));
+                return false;
+            }
+            self.token_owner.insert(id, caller);
+            let balance = self.balance_of_or_zero(caller);
+            self.owned_tokens_count.insert(caller, balance + 1);
+            self.env().emit_event(Transfer {
+                from: None,
+                to: caller,
+                id: id,
+            });
+            true
+        }
+
+        /// Get token balance of specific account.
+        #[ink(message)]
+        fn balance_of(&self, owner: AccountId) -> Balance {
+            let balance = self.balance_of_or_zero(owner);
+            self.env().println(&format!("Erc721::balance_of(owner = {:?}) = {:?}", owner, balance));
+            balance
+        }
+
+        /// Get owner for specific token.
+        #[ink(message)]
+        fn owner_of(&self, id: Hash) -> AccountId {
+            let owner = self.owner_of_or_none(id);
+            self.env().println(&format!("Erc721::owner_of(token = {:?}) = {:?}", id, owner));
+            owner
+        }
+
+        /// The approved address for this token, or the none address if there is none
+        #[ink(message)]
+        fn get_approved(&self, id: Hash) -> AccountId {
+            let account = self.approved_of_or_none(id);
+            account
+        }
+
+        /// Sets or unsets the approval of a given operator to transfer all tokens of caller
+        #[ink(message)]
+        fn set_approval_for_all(&mut self, to: AccountId, approved: bool) -> bool {
+            let caller = self.env().caller();
+            let hashmap = self.operator_approvals.get_mut(&caller);
+            match hashmap {
+                Some(hashmap) => {
+                    hashmap.insert(to, approved);
+                },
+                None => {
+                    unsafe {
+                        use ink_core::storage::alloc::AllocateUsing;
+                        use ink_core::storage::alloc::Initialize;
+                        use ink_core::storage::alloc::BumpAlloc;
+                        use ink_core::storage::Key;
+                        let mut alloc = BumpAlloc::from_raw_parts(Key([0x0; 32]));
+                        let mut default_hashmap: storage::HashMap<AccountId, bool> = storage::HashMap::allocate_using(&mut alloc).initialize_into(());
+                        default_hashmap.insert(to, approved);
+                        self.operator_approvals.insert(caller, default_hashmap);
+                    }
```

Comment:
Basically this whole thing is screwed but also can be easily solved by just using `storage::HashMap<(AcocuntId, AccountId), bool>` as described above.